### PR TITLE
feat: enable MediaStreams to handle WebRTC sound streams

### PR DESCRIPTION
**Pull request by menduz**
We had a hack for this.

```ts
function connectStream(stream: MediaStream){
  const sound = new BABYLON.Sound('avatar-stream', null, scene, null, {
   streaming: true,
   spatialSound: true
  })

  sound['_htmlAudioElement'] = new Audio()
  sound['_isReadyToPlay'] = true
  audioElement.srcObject = stream

  document.body.appendChild(audioElement)

  sound.play()

  return sound
}
```

This should enable us to use native WebRTC streams directly

```ts
function connectStream(stream: MediaStream){
  return new BABYLON.Sound('avatar-stream', stream, scene, null, {
   streaming: true,
   autoplay: true,
   spatialSound: true
  })
}
```

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -43,5 +43,6 @@
         "**/*.d.ts": true,
         "assets": true
     },
+    "editor.tabSize": 4,
```
Comment (deltakosh) [inline .vscode/settings.json]: nope :)

File: src/Audio/babylon.sound.ts [hunk 2]
```diff
@@ -98,11 +98,25 @@ module BABYLON {
                         if (typeof (urlOrArrayBuffer) === "string") this._urlType = "String";
                         if (Array.isArray(urlOrArrayBuffer)) this._urlType = "Array";
                         if (urlOrArrayBuffer instanceof ArrayBuffer) this._urlType = "ArrayBuffer";
+                        if (urlOrArrayBuffer instanceof MediaStream) this._urlType = "MediaStream";
```
Comment (deltakosh) [inline src/Audio/babylon.sound.ts]: should use else here

File: .vscode/settings.json (hunk 1 above)
Comment (menduz) [reply .vscode/settings.json]: Maybe this should not be part of this PR. Several files seem to use four spaces. Does it make sense?

Comment (deltakosh) [reply .vscode/settings.json]: Well actually you are right this is ok just to have everything aligned

File: src/Audio/babylon.sound.ts (hunk 2 above)
Comment (menduz) [reply src/Audio/babylon.sound.ts]: I don't agree. If we remove the `if` and replace it with an `else` statement that would make `_urlType = MediaStream` in the case of not handled types like `boolean`.

By default `_urlType` is `"Unknown"`. That counts for `else`

Comment (menduz) [reply src/Audio/babylon.sound.ts]: Ohh. Should I add that to the rest of the `if` statements too?

Comment (deltakosh) [reply src/Audio/babylon.sound.ts]: Yep please:)

**Comment by menduz** [comment]
@deltakosh please review

**Comment by deltakosh** [comment]
Please just update the what’s new in dist/preview release?

**Comment by menduz** [comment]
Done

**Comment by deltakosh** [comment]
You rock!!

**Comment by menduz** [comment]
🎉

**Pull request merged**